github.go

```diff
@@ -93,34 +100,13 @@ func NewGitHubClient(source Source) (*GitHubClient, error) {
 
 	return &GitHubClient{
 		client:      client,
+		clientV4:    clientV4,
 		owner:       owner,
 		repository:  source.Repository,
 		accessToken: source.AccessToken,
 	}, nil
 }
 
-func (g *GitHubClient) ListReleases() ([]*github.RepositoryRelease, error) {
```

Comment:
I tested out the PR without setting an API key, and was getting the following error:

```
error running command: non-200 OK status code: 401 Unauthorized body: "{\"message\":\"This endpoint requires you to be authenticated.\",\"documentation_url\":\"https://docs.github.com/v3/#authentication\"}"
```

It seems like the GraphQL endpoint requires authentication,  even for public repos (see also https://github.com/r-lib/devtools/issues/1844). Using the GraphQL endpoints is definitely beneficial for `ListReleases`, but perhaps we should keep the legacy v3 API calls around in case the user doesn't have an access token set.

So, `ListReleases` could call the GraphQL client if the access token is set, and otherwise call the v3 API client.